Health System logic

JacopoG opened the pull request:
Hi,

I'm working on issue #6, this pull request is just to get some feedback on the logic.

I tried to implemented the logic in the simplest way, so I used the Signal/Listener API offered by Ashley. The Health System receive an ExplosionHitsPlayerEvent and it update the health. The event contains the player entity. Basically I delegated all the collision logic outside.

Other things to consider:
- [ ] health points default value
- [ ] bomb damage value and where to put it

JacopoG committed:
add HealthSystem skeleton

DeimosA commented on core/src/no/group15/playmagic/ecs/components/HealthComponent.kt:
```diff
@@ -0,0 +1,7 @@
+
+import com.badlogic.ashley.core.Component
+
+class HealthComponent : Component {
```
Should be poolable

DeimosA commented on core/src/no/group15/playmagic/ecs/components/HealthComponent.kt:
```diff
@@ -0,0 +1,7 @@
+import com.badlogic.ashley.core.Component
+
+class HealthComponent : Component {
+	var points : Int = 100 // TODO chose a good default value?
```
Max health can be stored as its own val

JacopoG committed:
healthComponent: add maxPoints, add interface Poolable

JacopoG commented:
@DeimosA I added the Poolable interface to the component (I have to do it also in other branches). I also added a maxPoints variable, check if that's what you meant!
